frontend/src/components/Page.js

```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useState } from "react";
+import { Switch, Route, BrowserRouter, Redirect } from "react-router-dom";
+
+import Navbar from "./Navbar"
+import Footer from "./Footer";
+import userContext from "../contexts/context";
+import Form from "./Form";
+import DishForm from "./Dish/DishForm";
+import DishIndex from "./Dish/DishIndex";
+
+function Page() {
+    const { currentUrl, setUrl } = useContext(userContext)
+    const [formInfo, setFormInfo] = useState({});
+    const [formCallback, setFormCallback] = useState("");
+
+    function dish() {
+      setUrl("/dish");
+    }
+
+    return (
+      <>
+        <Navbar />
+        <button onClick={dish}>Click dish</button>
+        <BrowserRouter>
```

Comment:
Now I understand the usage of react router. I'm not sure it's worth it though - the alternative would be to do:
```
        const context = useContext(userContext);
        let url = context.currentUrl;
        return (
             <>
              <NavBar/>
             <Page>
                 {url === '/dish' && <DishIndex .../>}
                 {url === '/form' && <Form .../>}
            </Page>
            </>
        )
```

Reply:
Both are ligitimate options, it's just cost/value thing - I'm not sure the router is actually making things more simple